File: 0219/main.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        
        num = dict()
        for i, n in enumerate(nums):
            if n in num:
                if i - num[n] > k:
                    num[n] = i
                else:
                    return True
            else:
                num[n] = i
        return False

File: 0219/main.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> maps;

        for(int i=0; i<nums.size(); ++i) {
            int& n = nums[i];
            if(maps.count(n)) {
                if(i - maps[n] <= k) {
                    return true;
                } else {
                    maps[n] = i;
                }
            } else {
                maps[n] = i;
            }
        }

        return false;
    }
};